### PR TITLE
Refactor Terraform module to support latest supervisor

### PR DESCRIPTION
**Pull request by reset**
This adds support for the latest supervisor to the Builder Terraform
module. It's been a while since we updated this so a lot has changed.

* Refactor all security groups to ensure only exactly the required
  ports are opened
* Refactor aws_instance definitions to place all services in separate
  instances. This can be changed at a later date, but we will use this
  for load testing purposes
* Remove variables which when set could cause Terraform to not run
  properly or build an unusable environment
* Simplified organization of resources into files describing the type
  of resources contained instead

tenor-52152408

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: terraform/security-groups.tf [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+resource "aws_security_group" "jobsrv" {
+  name        = "builder-jobsrv-${var.env}"
+  description = "Builder JobSrv instance"
+  vpc_id      = "${var.aws_vpc_id}"
+
+  ingress {
+    from_port = 5566
+    to_port   = 5567
```
Comment (chefsalim) [inline terraform/security-groups.tf]: Do we need the to_port to be 5569 to reflect the new logs port?

**Comment by mwrock** [comment]
@thesentinels approve

File: terraform/variables.tf [hunk 2]
```diff
@@ -51,22 +61,10 @@ variable "private_subnet_id" {
...
-
 variable "router_count" {
   description = "Number of RouteSrv to start"
 }
 
-variable "service_count" {
-  description = "Number of service servers to start"
-}
-
 variable "jobsrv_worker_count" {
```
Comment (chefsalim) [inline terraform/variables.tf]: This doesn't seem to be used anywhere - nuke it?

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/230942720).

File: terraform/variables.tf [hunk 3]
```diff
@@ -51,22 +61,10 @@ variable "private_subnet_id" {
...
 
-variable "monolith_count" {
-  description = "Number of monolithic servers to start"
-}
-
-variable "rest_api_count" {
-  description = "Number of Builder REST API servers to start"
-}
-
 variable "router_count" {
```
Comment (chefsalim) [inline terraform/variables.tf]: This doesn't seem to be used anywhere - nuke it?

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/230942720).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**